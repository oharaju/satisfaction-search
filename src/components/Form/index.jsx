import React, { useState } from 'react';
import Content from '@/components/Content';
import TextArea from '@/components/TextArea';
import Radio from '@/components/Radio';
import Select from '@/components/Select';
import Checkbox from '@/components/Checkbox';
import StarReview from '@/components/StarReview';

const json = {
  "itens":[
      {
          "typeQuestion":1,
          "answerValue":5,
          "mandatory":true,
          "content":"Também é importante ter um espaço para o dono da loja colocar uma descrição da pergunta para ajudar o entendimento do usuário"
      },
      {
          "typeQuestion":2,
          "answerValue":9,
          "mandatory":true,
          "content":"Também é importante ter um espaço para o dono da loja colocar uma descrição da pergunta para ajudar o entendimento do usuário"
      },
      {
          "typeQuestion":3,
          "mandatory":false,
          "content":"Descreva o motivo de sua avaliação"
      },
      {
          "typeQuestion":4,   
          "content":"Qual loja você frequenta?",
          "mandatory":true,
          "itens": [
              {
                  "value":0,
                  "description":"Opção 1"
              },
              {
                  "value":1,
                  "description":"Opção 2"
              },
              {
                  "value":2,
                  "description":"Opção 3"
              }
          ]
      },
      {
          "typeQuestion":5,   
          "content":"Pergunta de escolha única?",
          "mandatory":true,
    "answerValue": 1,
           "itens": [
              {
                  "value":0,
                  "description":"Sim"
              },
              {
                  "value":1,
                  "description":"Não"
              }              
          ]
      },
       {
          "typeQuestion":6,   
          "horizontal":true,
          "content":"Pergunta de múltipla escolha?",
          "mandatory":true,
           "itens": [
              {
                  "value":0,
                  "description":"Opção 1"
              },
              {
                  "value":1,
                  "description":"Opção 2"
              },
              {
                  "value":2,
                  "description":"Opção 3"
              },
              {
                  "value":3,
                  "description":"Opção 4"
              }
          ]
      },
       {
          "typeQuestion":6,   
          "horizontal":false,
          "mandatory":true,
          "content":"Pergunta de múltipla escolha?",
           "itens": [
              {
                  "value":0,
                  "description":"Opção 1"
              },
              {
                  "value":1,
                  "description":"Opção 2"
              },
              {
                  "value":2,
                  "description":"Opção 3"
              }             
          ]
      },
      {
          "typeQuestion":3,
          "answerValue":"Exemplo de resposta",
          "mandatory":false,
          "content":"Pergunta de texto?"
      }
  ],
  "error":"",
  "warning":""
}

const itemsStars= [... new Array(5).keys()];

export default function Form() {

  const [activeStar, setActiveStar] = useState();

  const onClickStar = (index) => {
    setActiveStar((oldState) => (oldState === index ? undefined : index));
  }

  return(
    <div className='bg-white p-8 h-full max-h-lvh'>
      {json.itens.map(({typeQuestion, content,...rest}, index) => {
        return (
          <div key={index}>
            {content && <Content>{content}</Content>}

            {typeQuestion === 1 && 
              itemsStars.map((index) => (
                <StarReview {...rest} key={index} isActive={index <= activeStar} onClick={() => onClickStar(index)} />)
            )}

            {typeQuestion === 3 && <TextArea key={index} />}
            {typeQuestion === 4 && <Select content={content} {...rest} key={index} />}
            {typeQuestion === 5 && <Radio {...rest} key={index} />}
            {typeQuestion === 6 && <Checkbox {...rest} key={index} />}
            <br/><br/>
          </div>
        )
      })}
    </div>
  )
}