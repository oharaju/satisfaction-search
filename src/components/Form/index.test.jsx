import renderer from 'react-test-renderer';
import React from 'react';
import Form from './index';

it('get json and display it on screen correctly', () => {
  const tree = renderer
    .create(<Form/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});