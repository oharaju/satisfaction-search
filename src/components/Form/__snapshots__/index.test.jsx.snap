// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get json and display it on screen correctly 1`] = `
<div>
  <p>
    Também é importante ter um espaço para o dono da loja colocar uma descrição da pergunta para ajudar o entendimento do usuário
  </p>
  <div>
    <button>
      star1
    </button>
    <button>
      star2
    </button>
    <button>
      star3
    </button>
    <button>
      star4
    </button>
    <button>
      star5
    </button>
    <input
      type="text"
      value={5}
    />
  </div>
  <br />
  <br />
  <p>
    Também é importante ter um espaço para o dono da loja colocar uma descrição da pergunta para ajudar o entendimento do usuário
  </p>
  <br />
  <br />
  <p>
    Descreva o motivo de sua avaliação
  </p>
  <textarea
    type="text"
  />
  <br />
  <br />
  <p>
    Qual loja você frequenta?
  </p>
  <select
    name="select"
  >
    <option
      selected={true}
    >
      Qual loja você frequenta?
    </option>
    <option
      value="valor3"
    >
      Valor 3
    </option>
    <option
      value="valor3"
    >
      Valor 3
    </option>
    <option
      value="valor3"
    >
      Valor 3
    </option>
  </select>
  <br />
  <br />
  <p>
    Pergunta de escolha única?
  </p>
  <div>
    <input
      name="fav_language"
      type="radio"
      value={0}
    />
    <label>
      Sim
    </label>
    <input
      name="fav_language"
      type="radio"
      value={1}
    />
    <label>
      Não
    </label>
  </div>
  <br />
  <br />
  <p>
    Pergunta de múltipla escolha?
  </p>
  <div>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={0}
    />
    <label>
      Opção 1
    </label>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={1}
    />
    <label>
      Opção 2
    </label>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={2}
    />
    <label>
      Opção 3
    </label>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={3}
    />
    <label>
      Opção 4
    </label>
  </div>
  <br />
  <br />
  <p>
    Pergunta de múltipla escolha?
  </p>
  <div>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={0}
    />
    <label>
      Opção 1
    </label>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={1}
    />
    <label>
      Opção 2
    </label>
    <input
      name="fav_languagecheckbox"
      type="checkbox"
      value={2}
    />
    <label>
      Opção 3
    </label>
  </div>
  <br />
  <br />
  <p>
    Pergunta de texto?
  </p>
  <textarea
    type="text"
  />
  <br />
  <br />
</div>
`;
